operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/GenericKubernetesResourceMatcher.java

```diff
@@ -61,6 +66,9 @@ public static <R extends HasMetadata> Result<R> match(R desired, R actualResourc
       var desiredSpecNode = objectMapper.valueToTree(ReconcilerUtils.getSpec(desired));
       var actualSpecNode = objectMapper.valueToTree(ReconcilerUtils.getSpec(actualResource));
       var diffJsonPatch = JsonDiff.asJson(desiredSpecNode, actualSpecNode);
+      if (strongEquality && diffJsonPatch.size() > 0) {
```

Comment:
A comment as to why this determines that things are not equal according to "strong equality" would be helpful. For that matter, defining what strong equality is would be helpful! 😄

Reply:
renamed to equality, added detailed javadoc